lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -97,6 +98,30 @@ def add_customer_data(request, options, payment)
         }.compact
       end
 
+      def add_custom_fields_data(request, options)
+        if options[:fingerprint].present?
```

Comment:
I think we should use `device_fingerprint` for the option name as well, as this pattern is used elsewhere.